src/Pact/Eval.hs

```diff
@@ -521,42 +521,68 @@ typecheck ps = void $ foldM tvarCheck M.empty ps where
                         evalError (_tInfo t) $ "Type error: values for variable " ++ show _aType ++
                         " do not match: " ++ show (prevTy,ty)
 
-check1 :: forall e . Info -> Type (Term Name) -> Term Name -> Eval e (Maybe (TypeVar (Term Name),Type (Term Name)))
-check1 i spec t = do
+-- | 'typecheckTerm i spec t' checks a Term 't' against a specified type 'spec'.
```

Comment:
(WARNING: Bikeshedding. Feel free to ignore)

What is a 'spec', though, in this case. It looks like we're comparing argument types and their occurences, so shouldn't we just use the same 'ty' suffix to make it easier? My suggests would be `argTy` and `occTy` for argument type and occurrence type.